### PR TITLE
CLN: remove _maybe_update_attributes

### PR DESCRIPTION
**Pull request by jbrockmendel**
It is only non-trivial for DTI and TDI, and only actually needed in `__array_wrap__`.  It is cleaner to implement the relevant patch in `DatetimeIndexOpsMixin.__array_wrap__` and avoid the need for maybe_update_attributes altogether.

### CODE REVIEW

File: pandas/core/arrays/timedeltas.py [hunk 1]
```diff
@@ -776,12 +776,14 @@ def __rdivmod__(self, other):
...
         return res1, res2
 
-    # Note: TimedeltaIndex overrides this in call to cls._add_numeric_methods
     def __neg__(self):
         if self.freq is not None:
             return type(self)(-self._data, freq=-self.freq)
         return type(self)(-self._data)
 
+    def __pos__(self):
+        return type(self)(self._data, freq=self.freq)
```
Comment (jorisvandenbossche) [inline pandas/core/arrays/timedeltas.py]: This is new? Or where was it done before?

File: pandas/core/indexes/datetimelike.py [hunk 2]
```diff
@@ -163,6 +164,20 @@ def values(self):
     def asi8(self):
         return self._data.asi8
 
+    def __array_wrap__(self, result, context=None):
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/datetimelike.py]: sidenote (not for this PR): we should move to `__array_ufunc__` here instead

File: pandas/core/arrays/timedeltas.py (hunk 1 above)
Comment (TomAugspurger) [reply pandas/core/arrays/timedeltas.py]: In `_add_numeric_methods_unary`.

**Review by TomAugspurger — APPROVED** [review]

**Comment by jreback** [comment]
can you rebase

File: pandas/core/indexes/datetimelike.py (hunk 2 above)
Comment (jreback) [reply pandas/core/indexes/datetimelike.py]: agreed on this, can you create an issue @jbrockmendel

**Comment by WillAyd** [comment]
Took a glance at the ASV portal this might have caused a lot of performance regressions:

https://pandas.pydata.org/speed/pandas/#/regressions

@jbrockmendel do you have a quick way of validating that?

**Comment by jbrockmendel** [comment]
> do you have a quick way of validating that?

Not a quick way, no.  I'll take a look, but likely won't get to it today.

**Comment by WillAyd** [comment]
Oh yea no rush - thanks!

**Comment by jbrockmendel** [comment]
@WillAyd looking at this it isn't obvious which regressions you think may have been caused here.  I see a bunch from August 15 linking to 15b11c83; are those the ones you have in mind?

**Comment by WillAyd** [comment]
Might have misread but thought that was the commit that preceded this one on master

**Comment by jorisvandenbossche** [comment]
It's indeed this commit that asv reports for the biggest regressions. But there was also a gap in asv runs between August 2 and this commit (August 15), so probably it can be anything in that window

**Comment by jorisvandenbossche** [comment]
@TomAugspurger it also seems that the last runs were in September?

**Comment by TomAugspurger** [comment]
Will check today. We had some issues with a bad ASV env failing builds.

**Pull request merged**